scripts/patch_inference.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+
+import argparse
+
+import netCDF4
+import numpy as np
+from tensorflow.keras.models import load_model
+
+"""
+Script to perform inference of a patchfile using a trained Tensorflow NN model
+
+Example Usage:
+#Random patches
+./patch_inference.py "/glade/scratch/gwallach/goes16/goes16ci_model_cpu2020-06-10 16:14:15.932711.h5" "/glade/scratch/gwallach/goes16_nc/ABI_patches_32/abi_patches_20190915T000000.nc" lightning_20190915T000000_out.nc
+#Midwest
+./patch_inference.py "/glade/scratch/gwallach/goes16/goes16ci_model_cpu2020-06-10 16:14:15.932711.h5" "/glade/p/ral/wsap/petzke/lightning/abi/patches/abi_patches_20190601T180000.nc" lightning_20190601T18000_out.nc
+"""
+
+
+def infer(modelfile,patchfile,outfile,verbose=False):
+    model = load_model(modelfile)
+    if verbose:
+        model.summary()
+
+    print("Loading data",flush=True)
+    nc = netCDF4.Dataset(patchfile)
+    abi_data = nc['abi'][:]
+    time_data = nc['time'][:]
+    lon_data = nc['lon'][:]
+    lat_data = nc['lat'][:]
+    patch_data = nc['patch'][:]
+
+    patch_dim_size = nc.dimensions['patch'].size
+    y_dim_size = nc.dimensions['y'].size
+    x_dim_size = nc.dimensions['x'].size    
+    nc.close()
+
+    abi_data = np.swapaxes(abi_data,1,3)
+    abi_data = np.swapaxes(abi_data,1,2)
+    if verbose:
+        print("abi_data.shape = %s, abi_data = %s" % (abi_data.shape,abi_data[[0,-1]]),flush=True)
+
+    print("Performing inference",flush=True)
+    y_hat = model.predict(abi_data)
+
+    y_hat = y_hat.reshape((len(y_hat),))
```

Comment:
Would it be possible to reshape the predictions into the shape of the GLM grid the patches are derived from instead of leaving it as a 1D vector?

Reply:
Sure, I've reshaped the output to match the dimensions and format of the GLM nc files.